lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -124,6 +136,39 @@ def get_course_modes(self, obj):
             for mode in course_modes
         ]
 
+    def to_representation(self, instance: CourseEnrollment) -> 'OrderedDict':  # noqa: F821
+        """
+        Override the to_representation method to add the course_status field to the serialized data.
+        """
+        data = super().to_representation(instance)
+
+        if 'course_progress' in self.context.get('requested_fields', []) and self.context.get('api_version') == API_V4:
+            data['course_progress'] = self.calculate_progress(instance)
+
+        return data
+
+    def calculate_progress(self, model: CourseEnrollment) -> Dict[str, int]:
```

Comment:
Thank you for annotating types! Three notes:

1. The return type for this function should be `Dict[int, int]`
2. The annotations in this app are not checked in CI. If you would like them to be automatically checked, you could add the mobile_api or mobile_api/users to [the allowlist in mypy.ini](https://github.com/openedx/edx-platform/blob/master/mypy.ini#L8). If not, that's OK, just be aware that you'll need to check your annotations by hand.
3. Totally optional, no need to make this change: since the platform is on Py3.11 now, you can just use the lowercase builtin collection names now. No need to import them from typing. For example, `dict[int, int]`, `list[str]`. And instead of  `Union[str, None]` you can do `str | None`. They have the same meaning, the new ones are just easier to write.

Reply:
Thank you for notes! Regarding the third point, we implemented this a while ago in Python 3.8. I agree that it's definitely worth changing it to a more convenient format now since the platform is on py3.11.

Reply:
I think we need to use `Dict[str, int]` typing here, because we have dictionary with the next structure:
```
{
            'total_assignments_count': 5,
            'assignments_completed': 10,
}
```
key -> str type
value -> int type

This is described in the following docs:
https://peps.python.org/pep-0484/#the-typing-module

Reply:
Yup, you're totally right, my mistake